cfme/tests/storage/test_volume.py

```diff
@@ -1,54 +1,68 @@
+import fauxfactory
 
-from cfme.exceptions import ItemNotFound
 from cfme import test_requirements
-from cfme.cloud.provider import CloudProvider
 from cfme.cloud.provider.openstack import OpenStackProvider
-
-from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.storage.volume import VolumeAllView
 
 
 pytestmark = [
     pytest.mark.tier(3),
     test_requirements.storage,
-    pytest.mark.usefixtures('openstack_provider', 'setup_provider'),
-    pytest.mark.provider([CloudProvider]),
+    pytest.mark.ignore_stream("upstream"),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([OpenStackProvider], scope='module'),
 ]
 
-
-@pytest.mark.uncollectif(lambda provider: not provider.one_of(OpenStackProvider))
-def test_volume_navigation(openstack_provider, appliance):
-    # grab a volume name, the table returns a generator so use next
-
-    collection = appliance.collections.volumes
-    view = navigate_to(collection, 'All')
-
-    try:
-        volume_name = view.entities.get_first_entity().name
-    except(StopIteration, ItemNotFound):
-        pytest.skip('Skipping volume navigation for details, no volumes present')
-
-    volume = collection.instantiate(name=volume_name, provider=openstack_provider)
-
-    assert view.is_displayed
-
-    view = navigate_to(volume, 'Details')
-    assert view.is_displayed
-
-    view = navigate_to(collection, 'Add')
-    assert view.is_displayed
-
-
-@pytest.mark.uncollectif(lambda provider: not provider.one_of(OpenStackProvider))
-def test_volume_collective_crud(openstack_provider, appliance):
-    collection = appliance.collections.volumes
-    view = navigate_to(collection, 'All')
-
-    volumes = [collection.instantiate(name=item.name, provider=openstack_provider)
-              for item in view.entities.get_all()]
-
-    if volumes:
-        collection.delete(*volumes)
-    else:
-        pytest.skip("Skipping volume collective deletion, no volumes present")
+STORAGE_SIZE = 1
+
+
+def test_storage_volume_create_cancelled_validation(appliance, provider):
+    """ Test Attach instance to storage volume cancelled
+
+    prerequisites:
+        * Storage provider
+
+    Steps:
+        * Navigate to storage add volume page
+        * Click Cancel button
+        * Assert flash message
+    """
+    volume_collection = appliance.collections.volumes
+    manager_name = '{} Cinder Manager'.format(provider.name)
+    volume_collection.create(name=fauxfactory.gen_alpha(),
+                             storage_manager=manager_name,
+                             tenant=provider.data['provisioning']['cloud_tenant'],
+                             size=STORAGE_SIZE,
+                             provider=provider,
+                             cancel=True)
+
+    view = volume_collection.create_view(VolumeAllView)
+    view.flash.assert_message('Add of new Cloud Volume was cancelled by the user')
+
+
+@pytest.mark.tier(1)
+def test_storage_volume_crud(appliance, provider):
```

Comment:
**QUESTION:** Since this is a a CRUD test, shouldn't we have an UPDATE component (i.e. we change an existing entity, perhaps the one we created)?

Reply:
@jamesooden right; we have `edit` method for volume (yes it support only change in name) but nice suggestion going to push